### DISCUSSION TITLE
Custom validate image file

### DISCUSSION THREAD

**Discussion by nemanjam** [colinhacks/zod#796] [Q&A] [answered]
I can't find code example how to custom validate file input and check image file type and file size? Similar to this Yup example:

https://www.youtube.com/watch?v=tYGTjxhzrqY

What I found isn't specific enough.

```typescript
export const schema = z.object({
  avatar: z.instanceof(File),
});
```

**Comment by scotttrinh** [comment, marked as answer]
@nemanjam 

You can use `refine` here, too:

```ts
export const schema = z.object({
  avatar: z.instanceof(File).refine((file) => file.size < 1024 ** 2); // or whatever
});
```

Note that there seems to be an issue with Zod mutating the passed in object, which we're discussing a bit over in https://github.com/colinhacks/zod/discussions/801, so you might want to subscribe over there, too.

**Reply by nemanjam** [reply]
What will error message be for such custom logic, and do you know how such message can be specified?

**Reply by StefanTerdell** [reply]
Same as the rest, `refine` accepts a [second parameter for customizing error messages](https://github.com/colinhacks/zod#arguments).
